### DISCUSSION TITLE
How to Request a Service from Another Security Server in X-Road

### DISCUSSION THREAD

**Discussion by pas-comme** [nordic-institute/X-Road#2293] [Q&A]
To directly request a service via the security server "ss1," a client uses the following command:

bash
Copier le code
curl -vvv -H "X-Road-Client: demo/ETAT/0011/PUBLIC" -i 'http://demo-ss1.lxd:8080/r1/demo/ETAT/0011/PUBLIC/CITIZENS/persons'
Now, if another client belonging to the security server "ss2" wants to access the service provided through "ss1," what command should they use?

Note: The request should pass through "ss2," then to "ss1," and finally to the service provider, with the response returning in the reverse order.

Could you please provide the necessary command and explain the required configurations?

**Comment by petkivim** [comment]
Hi @pas-comme ! This is the command to access the `demo/ETAT/0011/PUBLIC/CITIZENS` service through `ss2`:

```bash
curl -vvv -H "X-Road-Client: demo/ETAT/0022/PRIVATE" -i 'http://demo-ss2.lxd:8080/r1/demo/ETAT/0011/PUBLIC/CITIZENS/persons'
```

The subsystem `demo/ETAT/0022/PRIVATE` that's used as a client in the request must be registered on `ss2`. In addition, the client subsystem `demo/ETAT/0022/PRIVATE` must have sufficient permissions to access the `demo/ETAT/0011/PUBLIC/CITIZENS` service on `ss1`. Therefore, you must update the service access permissions on `ss1`. After that you're ready to invoke the service using the `demo/ETAT/0022/PRIVATE` client subsystem through `ss2`.

Please note that `demo/ETAT/0022/PRIVATE` is just on example and you can replace it with any other client subsystem that you want.
